app/controllers/comments_controller.rb

```diff
@@ -87,18 +87,26 @@ def check_if_restricted
     redirect_to new_user_session_path(restricted_commenting: true)
   end
 
-  # Check to see if the ultimate_parent is a Work, and if so, if it allows
+  # Check to see if the ultimate_parent is a Work or AdminPost, and if so, if it allows
   # comments for the current user.
   def check_parent_comment_permissions
     parent = find_parent
-    return unless parent.is_a?(Work)
+    if parent.is_a?(Work)
+      entity = "work"
```

Comment:
You can call it [`translation_key`](https://guides.rubyonrails.org/v5.1/i18n.html#looking-up-translations).